src/transformers/pipelines.py

```diff
@@ -977,7 +980,11 @@ def __call__(self, *args, **kwargs):
             If ``self.return_all_scores=True``, one such dictionary is returned per label.
         """
         outputs = super().__call__(*args, **kwargs)
-        scores = np.exp(outputs) / np.exp(outputs).sum(-1, keepdims=True)
+
+        if self.model.config.num_labels == 1:
+            scores = 1 / (1 + np.exp(-outputs))
```

Comment:
Using `float` instead of `int` in the division to avoid casting.

```suggestion
            scores = 1.0 / (1.0 + np.exp(-outputs))
```